caddyhttp/httpserver/server.go

```diff
@@ -356,6 +356,16 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	c := context.WithValue(r.Context(), OriginalURLCtxKey, urlCopy)
 	r = r.WithContext(c)
 
+	// Setup a response recorder for this request.
+	rr := NewResponseRecorder(w)
+	w = rr
```

Comment:
I am not confident that this will have no impact on performance. A ResponseRecorder is necessary for implementing the response placeholders, but these are not always needed. Can we only attach the ResponseRecorder in middlewares that need it?

Reply:
Hmm, yeah, that's a fair point. The reason I set this up globally was that the replacer takes a  recorder as a parameter, and I couldn't seem to find a good way to do it ad-hoc within the middleware chain, since it requires re-defining the `*http.Request` being propagated. I'll take another pass at this to see if there are other options.